src/sentry/static/sentry/app/views/organizationIntegrations/integrationRow.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import Link from 'app/components/links/link';
+import {PanelItem} from 'app/components/panels';
+import PluginIcon from 'app/plugins/components/pluginIcon';
+import space from 'app/styles/space';
+import {Organization, SentryApp} from 'app/types';
+import {t} from 'app/locale';
+
+import IntegrationStatus from './integrationStatus';
+
+type Props = {
+  organization: Organization;
+  type: 'plugin' | 'firstParty' | 'sentryApp';
+  slug: string;
+  displayName: string;
+  status: 'Installed' | 'Not Installed' | 'Pending';
+  publishStatus: 'unpublished' | 'published' | 'internal';
+  configurations: number;
+};
+
+const urlMap = {
+  plugin: 'plugins',
+  firstParty: 'integrations',
+  sentryApp: 'sentry-apps',
+};
+
+const IntegrationRow = (props: Props) => {
+  const {
+    organization,
+    type,
+    slug,
+    displayName,
+    status,
+    publishStatus,
+    configurations,
+  } = props;
+
+  const baseUrl = `/settings/${organization.slug}/${urlMap[type]}/${slug}/`;
+
+  const renderDetails = () => {
+    if (type === 'sentryApp') {
+      return publishStatus !== 'published' && <PublishStatus status={publishStatus} />;
+    }
+    return configurations > 0 ? (
+      <StyledLink to={`${baseUrl}?tab=configurations`}>{`${configurations} Configuration${
+        configurations > 1 ? 's' : ''
+      }`}</StyledLink>
+    ) : null;
+  };
+
+  return (
+    <PanelItem p={0} flexDirection="column" data-test-id={slug}>
+      <FlexContainer>
+        <PluginIcon size={36} pluginId={slug} />
+        <Container>
+          <IntegrationName to={baseUrl}>{displayName}</IntegrationName>
+          <IntegrationDetails>
+            <IntegrationStatus status={status} />
+            {renderDetails()}
+          </IntegrationDetails>
+        </Container>
+      </FlexContainer>
+    </PanelItem>
+  );
+};
+
+const FlexContainer = styled('div')`
+  display: flex;
+  align-items: center;
+  padding: ${space(2)};
+`;
+
+const Container = styled('div')`
+  flex: 1;
+  padding: 0 16px;
+`;
+
+const IntegrationName = styled(Link)`
+  font-weight: bold;
+  color: ${props => props.theme.textColor};
+`;
+
+const IntegrationDetails = styled('div')`
+  display: flex;
+  align-items: center;
+  margin-top: 6px;
+  font-size: 0.8em;
+`;
+
+const StyledLink = styled(Link)`
+  color: ${p => p.theme.gray2};
+  &:before {
+    content: '|';
+    color: ${p => p.theme.gray1};
+    margin-right: ${space(0.75)};
+    font-weight: normal;
+  }
+`;
+
+type PublishStatusProps = {status: SentryApp['status']; theme?: any};
+
+const PublishStatus = styled(({status, ...props}: PublishStatusProps) => {
+  return <div {...props}>{t(`${status}`)}</div>;
```

Comment:
Is it safe to spread `props` directly on a HTML element? What if something non-standard is passed in, won't it error in console?

Reply:
@leeandher Totally safe. We know what props are being passed by `PublishStatusProps`, so the only prop that's being passed is `theme` which I believe is necessary. @NisanthanNanthakumar can confirm. And if we pass something that doesn't belong, we will see an error at compile-time for TS. That's why TS is so great :)

Reply:
@scefali @leeandher Yes, `theme` is necessary or the proper styles don't get applied.